internal/services/deposit/main.go

```diff
@@ -0,0 +1,329 @@
+package deposit
+
+import (
+	"context"
+	"github.com/GeoDB-Limited/odin-deposit-ether-svc/internal/config"
+	"github.com/GeoDB-Limited/odin-deposit-ether-svc/internal/data/system-contracts/generated"
+	"github.com/GeoDB-Limited/odin-deposit-ether-svc/odin/client"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"math/big"
+)
+
+const (
+	ETHDepositType = iota
+	ERC20DepositType
+)
+
+// Service defines a service that allows exchanging ETH and ERC20 for odin.
+type Service struct {
+	config   config.Config
+	log      *logrus.Logger
+	eth      *ethclient.Client
+	contract *generated.Bridge
+	odin     client.Client
+}
+
+// WithdrawalDetails defines withdrawal options
+type WithdrawalDetails struct {
+	EthereumAddress common.Address
+	OdinAddress     string
+	DepositAmount   *big.Int // amount of user's deposit
+	TokenAddress    common.Address
+	TokenSymbol     string
+	DepositType     int
+}
+
+// New creates a service that allows exchanging ETH and ERC20 for odin.
+func New(cfg config.Config) *Service {
+	odinClient := client.New(cfg)
```

Comment:
read comments about `WithSigner()` method and use it like: `odinClient := client.New(cfg).WithSigner()`